doc/sphinx/language/extensions/compil-steps.rst

```diff
@@ -0,0 +1,103 @@
+.. _compil-steps:
+
+=============================
+How Coq Processes Input Terms
+=============================
+
+It is in theory possible to write down every Gallina term explicitly
+as described in the :ref:`Core Language <core-language>` part of this
+manual, for instance
+:n:`Nat.add (S O) (Nat.mul (S (S O)) (S (S O)))`. However, this would
+be very tedious and error-prone and takes us away from our usual
+mathematical practice. To circumvent this, Rocq offers multiple
+preprocessing mechanisms to help fill the gap between what the users
+would like to input to the system and the fully formal core language
+expected by the kernel. We give an overview of all these steps below.
+
+For instance, the notation mechanisms reflects the eponymous mathematical
+practice and allows to write :n:`1 + 2 * 2` instead of the above
+term. Those mechanisms range from simple :ref:`Abbreviations` to full
+fledged :ref:`Notations` with user defined :ref:`syntaxes
+<ReservingNotations>`. Multiple interpretations can be given to the
+same syntax in different contexts thanks to the :ref:`scope
+<Scopes>` mechanism. For instance :n:`(1 + 2 * 2)%N` can be
+the above natural number expression while :n:`(1 + 2 * 2)%Z` can be
+an expression with integers.
+
+In order to take the best part of all these preprocessing mechanisms,
+one needs a basic understanding of the multiple steps needed to
+transform an inputed term (possibly with notations) into the valid
+Gallina term which Rocq will ultimately use internally. Terms given as input
+to Rocq go through several successive steps:
```

Comment:
```suggestion
In order to take the best part of all these preprocessing mechanisms,
one needs a basic understanding of the multiple steps needed to
transform an inputed term (possibly with notations) into the valid
Gallina term which Rocq will ultimately use internally. Terms given as input
to Rocq go through several successive steps:
```

The first sentence adds very little.  We wouldn't bother to explain the several successive steps unless it was important to the reader of this introductory material.  And I think distracting.

Correct spelling is "inputted", BTW.

"valid Gallina term" - doesn't fit `:uconstr(...)` so accurately

"successive" is a little misleading here, eg lexing and parsing a single sentence are overlapped in time.

Reply:
Well, this is technically true, but only in order to detect the end of the sentence. So the approximation is probably good enough for most users. Plus it's definitively a useful one. At worst we could add a note precising that, but that may be too much detail to state something relatively useless and obvious for those who really need to know it.